test/extended/util/metrics.go

```diff
@@ -0,0 +1,352 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+
+	"k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	kapierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubernetes/pkg/kubectl/util/term"
+	"k8s.io/kubernetes/test/utils"
+)
+
+// pods whose metrics show a larger ratio of requests per
+// second than maxQPSAllowed are considered "unhealthy".
+const (
+	maxQPSAllowed              = 1.5
+	bearerTokenCreationTimeout = 10 * time.Second
+)
+
+var (
+	// TODO: these exceptions should not exist. Update operators to have a better request-rate per second
+	perComponentNamespaceMaxQPSAllowed = map[string]float64{
+		"openshift-apiserver-operator":                            3.0,
+		"openshift-kube-apiserver-operator":                       6.5,
+		"openshift-kube-controller-manager-operator":              2.0,
+		"openshift-cluster-kube-scheduler-operator":               1.8,
+		"openshift-cluster-openshift-controller-manager-operator": 1.7,
+	}
+	calculatedBearerToken = ""
```

Comment:
not excited about the global var - could you just get the token at the top level place and pass it around?

Reply:
This still holds.